core/Application/Environment.php

```diff
@@ -115,7 +120,10 @@ private function createContainer()
     protected function getGlobalSettingsCached()
     {
         if ($this->globalSettingsProvider === null) {
-            $this->globalSettingsProvider = $this->getGlobalSettings();
+            $result = $this->invokeFirstManipulator(
```

Comment:
I'd suggest this approach:

```php
    protected function getGlobalSettingsCached()
    {
        if (null !== $this->globalSettingsProvider) {
            return $this->globalSettingsProvider;
        }
        
        $this->globalSettingsProvider = $this->invokeFirstManipulator(
            'makeKernelObject',
            array(
                'Piwik\Application\Kernel\GlobalSettingsProvider',
                $this->getKernelObjectsArray()
            )
        );
        
        if (! $this->globalSettingsProvider instanceof GlobalSettingsProvider) {
            $this->globalSettingsProvider = $this->getGlobalSettings();
        }
        
        return $this->globalSettingsProvider;
    }
```

This way, you're checking if `globalSettingsProvider` is always a instance of GlobalSettingsProvider, and the code is  clearer.